### PR TITLE
dev-python/rstcheck: Update to release 6.1.1

### PR DESCRIPTION
**Pull request by samuelbernardo**
Apply review suggestions from PR
https://github.com/gentoo/gentoo/pull/16399

Closes: https://bugs.gentoo.org/893326

### CODE REVIEW

**Comment by jonasstein** [comment]
if you add
`Closes: https://bugs.gentoo.org/893326`
to the git comment, we will not forget to close the ticket.

**Comment by samuelbernardo** [comment]
> if you add `Closes: https://bugs.gentoo.org/893326` to the git comment, we will not forget to close the ticket.

Done

File: dev-python/rstcheck/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<pkgmetadata>
+  <maintainer type="person">
+    <email>samuelbernardo.mail@gmail.com</email>
+    <name>Samuel Bernardo</name>
+  </maintainer>
+  <maintainer type="project">
+    <email>proxy-maint@gentoo.org</email>
+    <name>Proxy Maintainers</name>
+  </maintainer>
+  <upstream>
```
Comment (mgorny) [inline dev-python/rstcheck/metadata.xml:12]: It seems like a reasonable package, please add python@ as co-maintainer.

File: dev-python/rstcheck/rstcheck-6.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=poetry
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit distutils-r1
+
+DESCRIPTION="Checks syntax of reStructuredText and code blocks nested within it"
+HOMEPAGE="https://github.com/rstcheck/rstcheck"
```
Comment (mgorny) [inline dev-python/rstcheck/rstcheck-6.1.1.ebuild:12]: Add pypi too.

File: dev-python/rstcheck/rstcheck-6.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=poetry
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit distutils-r1
```
Comment (mgorny) [inline dev-python/rstcheck/rstcheck-6.1.1.ebuild:9]: Use pypi.eclass.

File: dev-python/rstcheck/rstcheck-6.1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+DESCRIPTION="Checks syntax of reStructuredText and code blocks nested within it"
+HOMEPAGE="https://github.com/rstcheck/rstcheck"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="dev-python/docutils[${PYTHON_USEDEP}]"
```
Comment (mgorny) [inline dev-python/rstcheck/rstcheck-6.1.1.ebuild:19]: Use the multi-line format for all dependencies to make diffs for future updates cleaner.

File: dev-python/rstcheck/rstcheck-6.1.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+"
+
+distutils_enable_tests pytest
+
+python_test() {
+	# Ignore the module from ${S}, use the one from ${BUILD_DIR}
+	# Otherwise, ImportMismatchError may occur
+	# https://github.com/gentoo/gentoo/pull/1622#issuecomment-224482396
+	# Override pytest options to skip flake8
+	pytest -vv --ignore=rst --override-ini="addopts=--doctest-modules" \
```
Comment (mgorny) [inline dev-python/rstcheck/rstcheck-6.1.1.ebuild:35]: epytest.

File: dev-python/rstcheck/rstcheck-6.1.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="dev-python/docutils[${PYTHON_USEDEP}]"
+BDEPEND="
+	>=dev-python/setuptools_scm-1.15.0[${PYTHON_USEDEP}]
+	test? (
+		${RDEPEND}
```
Comment (mgorny) [inline dev-python/rstcheck/rstcheck-6.1.1.ebuild:23]: distutils_enable_tests already does that.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by samuelbernardo** [comment]
Just upgraded to 6.2.0 in guru... I need to update ebuild with suggested changes from here and create a new PR.

**Pull request closed without merging**